### ISSUE TITLE
Clicking "Request and Install Certificate" Displays Server Error Page with "An existing connection was forcibly closed by the remote host"

### ISSUE THREAD

**Issue by sahgilbert** [sjkp/letsencrypt-siteextension#212]
Hello,

Thanks for creating the site extension, it's very clean and easy to use.

I've set everything up and followed all the instructions. I have successfully got to the page _https://example-site-prod.scm.azurewebsites.net/letsencrypt/Home/Install_ where I can click "Request and Install Certificate". When I do this, I select both my hostnames (www.examplesite.com, examplesite.com). I also add my email address and DO NOT click the "useStaging" checkbox, as it's for production. (See attached screenshot below).

<img width="1317" alt="lets encrypt - site extension screenshot 6 for github issue" src="https://user-images.githubusercontent.com/13452745/40184088-46eaaf08-59e7-11e8-99ff-fb55fa1dbfa3.png">

Upon clicking the button, I am redirected to _https://example-site-prod.scm.azurewebsites.net/letsencrypt/Home/install_

This page simply displays a standard Asp.Net Server Error page, with the following - 

**Server Error in '/letsencrypt' Application.
An existing connection was forcibly closed by the remote host.**
(See attached screenshot for details)
<img width="1245" alt="lets encrypt - site extension error for github issue" src="https://user-images.githubusercontent.com/13452745/40176423-38a24622-59d3-11e8-9118-af781a959954.png">

It is worth point out that before I get to the page for requesting the certificate, I arrive on this page - _https://example-site-prod.scm.azurewebsites.net/letsencrypt/Home/Hostname_

I have noticed that on this page, all of my Hostname SSL Bindings have their SslState set to DISABLED. Could this be something to do with it? Or is this simply because they do not yet have a TLS certificate attached to them? (See attached screenshot).

<img width="1193" alt="lets encrypt - site extension screenshot 2 for github issue" src="https://user-images.githubusercontent.com/13452745/40178076-5a226ef8-59d8-11e8-81ab-86c356cee530.png">

Another bit that I cannot understand, is why I can see that when navigating to _https://example-site-prod.scm.azurewebsites.net_ it shows HTTPS and "Secure", as if a certificate is bound to it? (Screenshot below).

<img width="418" alt="lets encrypt - site extension screenshot 3 for github issue" src="https://user-images.githubusercontent.com/13452745/40178330-14cd543e-59d9-11e8-8fac-f211a6af1bda.png">

You will also note that if I go to _examplesite.com_ it shows it as being "Not Secure", despite this custom domain being assigned to my underlying azurewebsite. (Screenshot below).

<img width="290" alt="lets encrypt - site extension screenshot 4 for github issue" src="https://user-images.githubusercontent.com/13452745/40178408-4f593424-59d9-11e8-9fde-5fdfdeb3c7ab.png">

Finally, if I go to _https://example-site-prod.azurewebsites.net/.well-known/acme-challenge/_ I see a completely blank page. (Screenshot below).

<img width="1435" alt="lets encrypt - site extension screenshot 5 for github issue" src="https://user-images.githubusercontent.com/13452745/40180370-8dfc09a4-59de-11e8-8a6f-eb53931e6e4f.png">

I have used the Kudu debug console and obtained the event log from the Azure Web App Service. (See below screenshot).

<img width="1436" alt="lets encrypt - site extension screenshot 7 for github issue" src="https://user-images.githubusercontent.com/13452745/40191516-b0c72d2c-59f9-11e8-8c95-ca241de54e63.png">

Also, using the Kudu debug console, and navigating to 
_D:\home\SiteExtensions\letsencrypt\config\httpsacme-v01.api.letsencrypt.org_ I can see that I have two files, a **"Registration"** and a **"Signer"**, but there are no other files in this folder. (See below screenshot).

<img width="1037" alt="lets encrypt - site extension screenshot 8 for github issue" src="https://user-images.githubusercontent.com/13452745/40192023-161f820e-59fb-11e8-9887-017da73d1fba.png">

How should I resolve this?

Thanks,
Simon

**Comment by sahgilbert** [comment]
After much thinking, I realised that I had already enforced HTTPS on the Azure Web App Service (specifically on the SSL Settings page). 

I had enforced HTTPS BEFORE generating the certificate (which of course, happens over HTTP), so the website was blocking the request of its own certificate.

By disabling HTTPS on the Azure Web App Service, and then requesting the certificate again, the certificate generation was successful.

<img width="1226" alt="lets encrypt - site extension screenshot 1 for github issue answer" src="https://user-images.githubusercontent.com/13452745/40231933-c8f01dfa-5a94-11e8-8b99-804a687e6a6c.png">

I then re-enabled HTTPS on the Azure Web App Service, and refreshed the website, resulting in the site showing as HTTPS now.

<img width="390" alt="lets encrypt - site extension screenshot 2 for github issue answer" src="https://user-images.githubusercontent.com/13452745/40231945-d8fa937e-5a94-11e8-858f-40feff9d26e8.png">

Thanks.
